### PR TITLE
Modify NUOPC cap to accept separate glc runoff fluxes

### PR DESCRIPTION
**Pull request by alperaltuntas**
This PR addresses issue #281 

Changes made:
- Added Forr_rofl_glc and Forr_rofi_glc import files corresponding the glc runoff fluxes via the rof component.
- Currently, these new fluxes get added to fluxes%lrunoff and fluxes%frunoff. 
- For now, these new fluxes are imported only if cesm_coupled is `.true.`. If EMC folks decide the import these fluxes in the same way in the future, the import statements can be taken out of the if-blocks.

### CODE REVIEW

**Comment by alperaltuntas** [comment]
@mvertens Could you confirm if this PR addresses the issue #288 ?

File: config_src/drivers/nuopc_cap/mom_surface_forcing_nuopc.F90 [hunk 1]
```diff
@@ -472,6 +474,16 @@ subroutine convert_IOB_to_fluxes(IOB, fluxes, index_bounds, Time, valid_time, G,
       fluxes%frunoff(i,j) = kg_m2_s_conversion * IOB%frunoff(i-i0,j-j0) * G%mask2dT(i,j)
     endif
 
+    ! add liquid glc runoff flux via rof
+    if (associated(IOB%lrunoff_glc)) then
```
Comment (gustavo-marques) [inline config_src/drivers/nuopc_cap/mom_surface_forcing_nuopc.F90:502]: We should keep these new fields separate in fluxes and not under `fluxes%lrunoff` and `fluxes%frunoff`. 

Suggestion:
fluxes%lrunoff_glc(i,j) = fluxes%lrunoff_glc(i,j) + kg_m2_s_conversion * IOB%lrunoff_glc(i-i0,j-j0) * G%mask2dT(i,j)

fluxes%frunoff_glc(i,j) = fluxes%frunoff_glc(i,j) + kg_m2_s_conversion * IOB%frunoff_glc(i-i0,j-j0) * G%mask2dT(i,j)

**Comment by gustavo-marques** [comment]
Thanks for this PR, @alperaltuntas! 

Are the changes in [src/initialization/MOM_shared_initialization.F90](https://github.com/NCAR/MOM6/pull/288/files#diff-c103a203ffaf1fc606cfef99535b7c8ae3373b191663b9bb8066da5572f33eb0) intended?

In addition to what I mentioned above, we also need to: 

1) Add two new fields for the heat content associated with lrunoff_glc and frunoff_glc ()
heat_content_cond

https://github.com/NCAR/MOM6/pull/214
2)

**Comment by gustavo-marques** [comment]
Thanks for this PR, @alperaltuntas! 

Are the changes in [src/initialization/MOM_shared_initialization.F90](https://github.com/NCAR/MOM6/pull/288/files#diff-c103a203ffaf1fc606cfef99535b7c8ae3373b191663b9bb8066da5572f33eb0) intended?

In addition to what I mentioned above, we also need to: 

1) Add two new fields for the heat content associated with lrunoff_glc and frunoff_glc (heat_content_lrunoff_glc and heat_content_frunoff_glc). The coupler should provide these fields, and [here](https://github.com/ESCOMP/CMEPS/pull/278/files) is a PR that can be used as an example of adding them in CMEPS. 

We should only account for `heat_content_lrunoff_glc` and `heat_content_frunoff_glc` when `ENTHALPY_FROM_COUPLER = True`.

2) The new mass and heat fluxes should be added to `src/core/MOM_forcing_type.F90` and `src/diagnostics/MOM_sum_output.F90`. Please follow what was done in https://github.com/NCAR/MOM6/pull/214

**Comment by alperaltuntas** [comment]
@gustavo-marques I just pushed a commit here and a PR to CMEPS (https://github.com/ESCOMP/CMEPS/pull/489) to address your comments. One outstanding task is to determine what to do with the diagnostics associated with `lrunoff` and `frunoff`. Now that we have separated `lrunoff_glc` from `lrunoff` and `frunoff_glc` from `frunoff`, I wonder whether we should split the diagnostics as well, and if so, whether the split diagnostics truly correspond to cmor definitions of those diagnostics. See, for instance, the `frunoff` diagnostic whose cmor name is set to be `ficeberg` as defined in `src/core/MOM_forcing_type.F90`

**Comment by gustavo-marques** [comment]
These are the new available diagnostics:

```
"frunoff_glc"  [Unused]
    ! modules: {ocean_model,ocean_model_d2}
    ! long_name: Frozen glc runoff (calving) and iceberg melt into ocean
    ! units: kg m-2 s-1
    ! standard_name: glc_water_flux_into_sea_water_from_icebergs
    ! cell_methods: xh:mean yh:mean area:mean
"lrunoff_glc"  [Unused]
    ! modules: {ocean_model,ocean_model_d2}
    ! long_name: Liquid runoff (glaciers) into ocean
    ! units: kg m-2 s-1
    ! standard_name: water_flux_into_sea_water_from_glaciers
    ! cell_methods: xh:mean yh:mean area:mean
"total_frunoff_glc"  [Unused]
    ! modules: ocean_model
    ! long_name: Area integrated frozen glc runoff (calving) & iceberg melt into ocean
    ! units: kg s-1
"total_lrunoff_glc"  [Unused]
    ! modules: ocean_model
    ! long_name: Area integrated liquid glc runoff into ocean
    ! units: kg s-1
    ! cell_methods: xh:mean yh:mean area:mean
"heat_content_frunoff_glc"  [Unused]
    ! modules: {ocean_model,ocean_model_d2}
    ! long_name: Heat content (relative to 0C) of solid glc runoff into ocean
    ! units: W m-2
    ! cell_methods: xh:mean yh:mean area:mean
"heat_content_lrunoff_glc"  [Unused]
    ! modules: {ocean_model,ocean_model_d2}
    ! long_name: Heat content (relative to 0C) of liquid glc runoff into ocean
    ! units: W m-2
```

**Review by gustavo-marques — COMMENTED** [review]
One minor suggestion: only make the new diagnostics available if GLC is active.

**Pull request merged**